plugins/arrow-flight-rpc/src/main/java/org/opensearch/arrow/flight/bootstrap/FlightClientManager.java

```diff
@@ -99,7 +100,19 @@ public FlightClientManager(
      * @return An OpenSearchFlightClient instance for the specified node
      */
     public Optional<FlightClient> getFlightClient(String nodeId) {
-        return Optional.ofNullable(flightClients.get(nodeId));
+        ClientHolder clientHolder = flightClients.getOrDefault(nodeId, null);
+        return clientHolder == null ? Optional.empty() : Optional.ofNullable(clientHolder.flightClient);
+    }
+
+    /**
+     * Returns the location of a Flight client for a given node ID.
+     *
+     * @param nodeId The ID of the node for which to retrieve the location
+     * @return The Location of the Flight client for the specified node
+     */
+    public Location getFlightClientLocation(String nodeId) {
```

Comment:
Please, use `Optional` here as well:

```suggestion
    public Optional<Location> getFlightClientLocation(String nodeId) {
```